autopass

```diff
@@ -222,29 +224,44 @@ class Entry
   end
 
   def window_id
-    output = `xdotool search #{Shellwords.escape(window_regex)}`.chomp
-    if output.lines.size == 1
-      output.to_i
-    else
-      notify('Please select the target window')
-      xwininfo = `xwininfo -int`
-      title = xwininfo.match(/Window id: \d+ "(?<title>.*)"$/)[:title]
-      print "selected window: `#{title}` "
-      IO.popen('xclip', 'w+') do |io|
-        io.write(title)
-        io.close_write
-      end
-      puts 'copied to clipboard'
-      xwininfo[/(?<=Window id: )\d+/]
-    end
+    # try window
+    output = search_window_id(attributes.window)
+    return output.to_i if output.lines.size == 1
+
+    # try url
+    output = search_window_id(Regexp.escape(attributes.url))
+    return output.to_i if output.lines.size == 1
+
+    # try window with browser
+    output = search_window_id("#{attributes.window}.*#{Regexp.union(BROWSERS)}")
```

Comment:
Why is the browser necessary here? i think it's problematic due to the BROWSERS constant not possibly containing all browsers out there.